### PR TITLE
Implement `unused_parens` for `const` and `static` items

### LINKED ISSUE #67942: Missing unnecessary parentheses warning in constant expressions
```
#![allow(dead_code, unused_variables)]
fn main() {
    const N1: f64 = (2.0 * 2.0);
    let n2: f64 = (2.0 * 2.0);
}
```
It doesn't give a warning for the unnecessary parentheses for N1:
```
warning: unnecessary parentheses around assigned value
 --> ...\test.rs:4:19
  |
4 |     let n2: f64 = (2.0 * 2.0);
  |                   ^^^^^^^^^^^ help: remove these parentheses
  |
  = note: `#[warn(unused_parens)]` on by default
  ```

This issue has been assigned to @Tyg13 via [this comment](https://github.com/rust-lang/rust/issues/67942#issuecomment-581078243).

### PR DESCRIPTION
**Pull request by Tyg13**
Fixes #67942

### CODE REVIEW

**Comment by Tyg13** [comment]
Ah, darn. Looks like there are some issues to be resolve with the submodule. I'll update it in a separate branch and resolve them there.

File: src/librustc_lint/unused.rs [hunk 1]
```diff
@@ -587,6 +587,16 @@ impl EarlyLintPass for UnusedParens {
             }
         }
     }
+
+    fn check_item(&mut self, cx: &EarlyContext<'_>, item: &ast::Item) {
+        use ast::ItemKind::*;
+        match item.kind {
+            Const(.., ref expr) | Static(.., ref expr) => {
```
Comment (Centril) [inline src/librustc_lint/unused.rs]: Use `if let` here instead.

**Comment by Tyg13** [comment]
Whoops, that's what I get for pushing without running `check`. Should be sorted now.

**Pull request merged**